src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -987,3 +988,10 @@ def type_string_to_type_object(type_str: str) -> DataType:
 ColumnOrSqlExpr = Union["snowflake.snowpark.column.Column", str]
 LiteralType = Union[VALID_PYTHON_TYPES_FOR_LITERAL_VALUE]
 ColumnOrLiteral = Union["snowflake.snowpark.column.Column", LiteralType]
+
+class CopyOptions(TypedDict, total=False):
+    overwrite: bool
+    single: bool
+    max_file_size: float
+    include_query_id: bool
+    detailed_output: bool
```

Comment:
I think value in string form is also allowed?
also are the keys case sensitive?

if I have a copy options looking like
```python
option_dict = {"OVERWRITE": "TRUE"}
```

would pylance complain?